File: src/_03_BehavioralPattern/_03_06_Memento/Game.js
import GameSave from "./GameSave.js";

export default class Game {
  /** @type { number } */
  _blueTeamScore;

  /** @type { number } */
  _redTeamScore;

  /** @returns { number } */
  getBlueTeamScore() {
    return this._blueTeamScore;
  }
  
  /** @param { number } blueTeamScore */
  setBlueTeamScore(blueTeamScore) {
    this._blueTeamScore = blueTeamScore;
  }

  /** @returns { number } */
  getRedTeamScore() {
    return this._redTeamScore;
  }

  /** @param { number } redTeamScore */
  setRedTeamScore(redTeamScore) {
    this._redTeamScore = redTeamScore;
  }

  /** @returns { GameSave } */
  save() {
    return new GameSave(this._blueTeamScore, this._redTeamScore);
  }

  /** @param { GameSave } gameSave  */
  restore(gameSave) {
    this._blueTeamScore = gameSave.getBlueTeamScore();
    this._redTeamScore = gameSave.getRedTeamScore(); 
  }
}

File: src/_03_BehavioralPattern/_03_06_Memento/java/after/Game.java
package _03_BehavioralPattern._03_06_Memento.java.after;

public class Game {

  private int blueTeamScore;

  private int redTeamScore;

  public int getBlueTeamScore() {
    return this.blueTeamScore;
  }
  public void setBlueTeamScore(int blueTeamScore) {
    this.blueTeamScore = blueTeamScore;
  }

  public int getRedTeamScore() {
    return this.redTeamScore;
  }
  public void setRedTeamScore(int redTeamScore) {
    this.redTeamScore = redTeamScore;
  }

  public GameSave save() {
    return new GameSave(this.blueTeamScore, this.redTeamScore);
  }

  public void restore(GameSave gameSave) {
    this.blueTeamScore = gameSave.getBlueTeamScore();
    this.redTeamScore = gameSave.getRedTeamScore();
  }
  
}
